src/server/metrics.rs

```diff
@@ -202,3 +206,164 @@ lazy_static! {
         .unwrap()
     };
 }
+
+// Defines a HdrHistogram, its minimum resolution value and maximum trackable time interval are about 500us and 1000s, respectively.
+macro_rules! define_hdrhistogram {
+    ($NAME: ident) => {
+        lazy_static! {
+            pub static ref $NAME: Arc<Mutex<HdrHistogram<u64>>> = Arc::new(Mutex::new(
+                HdrHistogram::<u64>::new_with_bounds(1 << 9, 1 << 30, 3).unwrap()
+            ));
+        }
+    };
+}
+
+define_hdrhistogram!(INVALID_HDR);
+define_hdrhistogram!(KV_GET_HDR);
+define_hdrhistogram!(KV_SCAN_HDR);
+define_hdrhistogram!(KV_PREWRITE_HDR);
+define_hdrhistogram!(KV_PESSIMISTIC_LOCK_HDR);
+define_hdrhistogram!(KV_PESSIMISTIC_ROLLBACK_HDR);
+define_hdrhistogram!(KV_COMMIT_HDR);
+define_hdrhistogram!(KV_CLEANUP_HDR);
+define_hdrhistogram!(KV_BATCH_GET_HDR);
+define_hdrhistogram!(KV_BATCH_ROLLBACK_HDR);
+define_hdrhistogram!(KV_SCAN_LOCK_HDR);
+define_hdrhistogram!(KV_RESOLVE_LOCK_HDR);
+define_hdrhistogram!(KV_GC_HDR);
+define_hdrhistogram!(KV_DELETE_RANGE_HDR);
+define_hdrhistogram!(RAW_GET_HDR);
+define_hdrhistogram!(RAW_BATCH_GET_HDR);
+define_hdrhistogram!(RAW_SCAN_HDR);
+define_hdrhistogram!(RAW_BATCH_SCAN_HDR);
+define_hdrhistogram!(RAW_PUT_HDR);
+define_hdrhistogram!(RAW_BATCH_PUT_HDR);
+define_hdrhistogram!(RAW_DELETE_HDR);
+define_hdrhistogram!(RAW_DELETE_RANGE_HDR);
+define_hdrhistogram!(RAW_BATCH_DELETE_HDR);
+define_hdrhistogram!(UNSAFE_DESTROY_RANGE_HDR);
+define_hdrhistogram!(COPROCESSOR_HDR);
+define_hdrhistogram!(COPROCESSOR_STREAM_HDR);
+define_hdrhistogram!(MVCC_GET_BY_KEY_HDR);
+define_hdrhistogram!(MVCC_GET_BY_START_TS_HDR);
+define_hdrhistogram!(SPLIT_REGION_HDR);
+define_hdrhistogram!(READ_INDEX_HDR);
+
+pub struct HdrTimer {
+    start: Instant,
+    hdr: Arc<Mutex<HdrHistogram<u64>>>,
+}
+
+impl HdrTimer {
+    pub fn new(hdr: &Arc<Mutex<HdrHistogram<u64>>>) -> Self {
+        Self {
+            hdr: hdr.clone(),
+            start: Instant::now_coarse(),
+        }
+    }
+
+    pub fn observe_duration(self) {
+        drop(self);
+    }
+
+    fn observe(&mut self) {
+        let v = self.start.elapsed().as_micros() as u64;
+        self.hdr.lock().unwrap().record(v).unwrap();
```

Comment:
I don't like calling lock every time in the so hot gRPC path.

Reply:
Unlike rust-prometheus' Histogram, this HdrHistogram does not provide concurrency support, its `record` method need use mutable borrow.